src/transport_picoquic.cpp

```diff
@@ -572,15 +594,47 @@ PicoQuicTransport::start()
     uint64_t current_time = picoquic_current_time();
 
     if (debug) {
-        debug_set_stream(stdout); // Enable picoquic debug
+        if (!opened_logging_fds)
+        {
+            // Open pipe for reading and writing
+            if (pipe(logging_fds) == 0) {
```

Comment:
Need to watch to ensure we don't introduce a blocking operation here.

Reply:
There ought not be one.  There is a separate thread reading the logs written into the pipe.